torch/testing/_internal/opinfo/definitions/linalg.py

```diff
@@ -1517,8 +1517,9 @@ def make_input():
         "linalg.lstsq",
         aten_name="linalg_lstsq",
         variant_test_name="grad_oriented",
-        # gradchecks for forward AD fails with multi-Tensor outputs
-        op=lambda a, b, driver: torch.linalg.lstsq(a, b, driver=driver)[0],
+        # gradchecks for forward AD fails with full output tuple
+        # works when taking [:2], which is (solution, residuals)
```

Comment:
This is expected, as the other two are non differentiable, so we can just remove this comment.

Reply:
I don't think it's expected. For example, `torch.linalg.lu` has non-differentiable output (`output_differentiability: [False, True, True]`), but `op=torch.linalg.lu` works in OpInfo. Also, for this function `torch.linalg.lstsq`, `[:3]` still works, which include one non-differentiable term.

Reply:
We work around this an similar problems (differentiability of complex svd and others) doing this:
https://github.com/pytorch/pytorch/blob/2068235c0addeadddb699b62c294dbbd33e24d51/torch/testing/_internal/opinfo/definitions/linalg.py#L1058-L1062

Which is equivalent to what you did, pretty much.

Reply:
I see.